test/hotspot/jtreg/runtime/Thread/TestAlwaysPreTouchStacks.java

```diff
@@ -103,83 +163,30 @@ public static void main(String[] args) throws Exception {
-                    // really working.
-                    if (preTouch && (double)committed < ((double)reserved * 0.75)) {
-                        throw new RuntimeException("Expected a higher ratio between stack committed and reserved.");
-                    } else if (!preTouch && (double)committed > ((double)reserved * 0.50)){
-                        throw new RuntimeException("Expected a lower ratio between stack committed and reserved.");
-                    }
-                    // Added sanity tests: we expect our test threads to be still alive when NMT prints its final
-                    // report, so their stacks should dominate the NMT-reported total stack size.
-                    long max_reserved = memoryCeilingMB * 3 * MB;
-                    long min_reserved = memoryCeilingMB * MB;
-                    if (reserved >= max_reserved || reserved < min_reserved) {
-                        throw new RuntimeException("Total reserved stack sizes outside of our expectations (" + reserved +
-                                                   ", expected " + min_reserved + ".." + max_reserved + ")");
-                    }
-                    foundLine = true;
-                    break;
-                }
-            }
-            if (!foundLine) {
-                throw new RuntimeException("Did not find expected NMT output");
-            }
-        }
-
+          ReservedCommitted pretouch_result = runPreTouchTest(true);
+          ReservedCommitted no_pretouch_result = runPreTouchTest(false);
+          if (pretouch_result.reserved == 0 || no_pretouch_result.reserved == 0) {
+            throw new RuntimeException("Could not run with PreTouch flag.");
+          }
+          double ratio_with = ((double)pretouch_result.committed) / pretouch_result.reserved;
+          double ratio_without = ((double)no_pretouch_result.committed) / no_pretouch_result.reserved;
+          System.out.println("ratio with PreTouch: " + ratio_with + " w/out: " + ratio_without);
+          if (ratio_without > 0.50) {
+            throw new RuntimeException("Expected a lower ratio between stack committed and reserved.");
+          }
+          if (ratio_with < ratio_without) {
+            throw new RuntimeException("Expected a higher ratio between stack committed and reserved.");
+          }
+          double estimated_stack_usage = 1 * MB;
+          double expected_stack_use_with_pretouch = 0.75 * threadStackSizeMB * MB;
+          if (ratio_with > expected_stack_use_with_pretouch) {
+            throw new RuntimeException("Expected a higher ratio of committed to reserved of stack with pretouch.");
+          }
+          double expected_delta = numThreads * (expected_stack_use_with_pretouch - estimated_stack_usage);
+          if ((pretouch_result.committed - no_pretouch_result.committed) < expected_delta) {
+            throw new RuntimeException("Expected a higher delta between stack committed of with and without pretouch.");
```

Comment:
Can you print out the expected values, please?

Reply:
Done.